cmd/ak/cmd/deployments/delete.go

```
@@ -20,7 +20,7 @@ var deleteCmd = common.StandardCommand(&cobra.Command{
 			return common.FailIfError(cmd, err, "deployment")
 		}
 
-		if err := common.FailIfNotFound(cmd, "deployment", d.IsValid()); err != nil { // test d != nil
+		if err := common.FailIfNotFound(cmd, "deployment", d.IsValid()); err != nil {
```

could you elaborate more on this FailIfNotFound method ? it is not very clear

---

it's not my invention actually :) There is a --fail flag. So cmd will finish with exit status 0, if the error is not found. Unless --fail flag is supplied. In this case cmd will report error if such an object isn't found

---

I understand that, but you're not really checking for not found, you check if d.IsValid which is really not readable 

can you just do 
```go
if !d.IsValid() {
  return errors.New("invalid deployment id")
}
```

---

this is due to new changes to sdk that were introduced. Previously all methods were returning nil, so it was clear and readable. Now, when not found, zero object will be returned and it should fail on `isValid()`. So actually failing on `IsValid` means not found

---

I understand why it should fail, I just think this is not readable to use fail if not found on not valid id

---

agree. But it's not something introduced by this PR. Actually it's something that was unreadable for me in the original sdk2 PR as well